src/sentry/api/bases/organization.py

```diff
@@ -355,20 +367,10 @@ def get_projects(
             force_global_perms=force_global_perms,
             include_all_accessible=include_all_accessible,
         )
-        filtered_project_ids = {p.id for p in filtered_projects}
-        filtered_project_slugs = {p.slug for p in filtered_projects}
-
-        if (
-            not include_all_accessible
-            and not filter_by_membership
-            and (
-                (ids and ids != filtered_project_ids) or (slugs and slugs != filtered_project_slugs)
-            )
-        ):
-            # If a user requests all projects - they should get back all projects they have permission for
-            # If a user requests specified projects, but they don't have access to them
-            # Then we should raise a permission denied
-            raise PermissionDenied
+
+        requesting_specific_projects = not include_all_accessible and not filter_by_membership
+        if requesting_specific_projects:
```

Comment:
super nit since we're not referencing the value again:
```suggestion
        if not include_all_accessible and not filter_by_membership:
```

Reply:
I actually purposefully did this for readability. It's not obvious that `not include_all_accessible and not filter_by_membership` means you're requesting specific projects. I considered combining them and adding a comment, but recently I've been trying to have the code itself readable instead of over relying on comments